python/adjoint/filters.py

```diff
@@ -36,6 +21,37 @@ def _centered(arr, newshape):
     return arr[tuple(myslice)]
 
 
+def _proper_pad(arr, pad_to):
```

Comment:
Would be good to add docstrings for all functions following these [guidelines](https://google.github.io/styleguide/pyguide.html#381-docstrings.